p2p/chain.py

```diff
@@ -569,10 +601,12 @@ def _handle_get_node_data(self, peer: ETHPeer, msg: List[Hash32]) -> None:
             nodes.append(node)
         peer.sub_proto.send_node_data(nodes)
 
-    async def _process_headers(self, peer: ETHPeer, headers: List[BlockHeader]) -> int:
+    async def _process_headers(
+            self, peer: HeaderRequestingPeer, headers_tuple: Tuple[BlockHeader, ...]) -> int:
         # This is needed to ensure after a state sync we only start importing blocks on top of our
         # current head, as that's the only one whose state root is present in our DB.
-        for header in headers.copy():
+        headers = list(headers_tuple)
```

Comment:
Maybe `headers` could be named `new_headers` to indicate that we're building the list of headers we don't already have.